metadata-ingestion/src/datahub/ingestion/source/sql/teradata.py

```diff
@@ -681,8 +681,10 @@ def __init__(self, config: TeradataConfig, ctx: PipelineContext):
 
     def _add_default_options(self, sql_config: SQLCommonConfig) -> None:
         """Add Teradata-specific default options"""
-        # Teradata does not support max_overflow, instead we use QueuePool when profiling
+        super()._add_default_options(sql_config)
         if sql_config.is_profiling_enabled():
+            # By default, Teradata uses SingletonThreadPool, which is not supported by sqlalchemy
```

Comment:
are there some github issues / slack overflow threads discussing this issue? 

would like to substantiate the "is not supported by sqlalchemy" bit - especially given that `SingletonThreadPool` comes from sqlalchemy https://docs.sqlalchemy.org/en/20/core/pooling.html#sqlalchemy.pool.SingletonThreadPool

I know that I'm being annoying about this, and that's intentional. See https://acryldata.notion.site/Chesterton-s-Fence-191fc6a64277801ba89edf5a118ab9b0?pvs=4 for why